numpy/core/src/npymath/npy_math_internal.h.src

```diff
@@ -530,19 +530,32 @@ npy_remainder@c@(@type@ a, @type@ b)
 
 NPY_INPLACE @type@
 npy_floor_divide@c@(@type@ a, @type@ b) {
-    @type@ div, mod;
+    @type@ mod;
     if (NPY_UNLIKELY(!b)) {
+        if(npy_isinf(a)) {
+            return NPY_NAN;
+        }
+
         /*
          * in2 == 0 (and not NaN): normal division will give the correct
          * result (Inf or NaN). `divmod` may set additional FPE for the modulo
          * evaluating to NaN.
          */
-        div = a / b;
+        return a / b;
     }
-    else {
-        div = npy_divmod@c@(a, b, &mod);
+
+    int is_inf_a = npy_isinf(a), is_inf_b = npy_isinf(b);
+
+    if(is_inf_a && !is_inf_b) {
+        return npy_copysign(NPY_INFINITY, a/b);
+
+    } else if(!is_inf_a && is_inf_b) {
+        return npy_copysign(0, a/b);
```

Comment:
I still think most of this returns the right thing in the normal branch already?

Reply:
the answer is right only if both dividend and divisor is +ve

Reply:
OK, right, I hadn't thought of changing the answer to `-0.0` from `-1.0`.  I suspect that is good, would be nice to mention explictly though.

What I meant was that:
```
In [1]: -43456666 / np.inf
Out[1]: -0.0
```
is already correct.  Yes, we probably want to avoid calling divmod.  But we can just use normal division and simplify those branches.

Reply:
ok, i have simplified the branch to use normal division if any of the divisor/dividend is inf